src/jbuild.ml

```diff
@@ -400,6 +402,7 @@ module Lib_dep = struct
           ; forbidden
           ; file
           }
+        | Template _
```

Comment:
We should have a message saying that variables are not allowed here. it will be clearer why this is rejected. Same remark for the subsequent cases in this PR